packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -152,23 +169,45 @@ class DoubleTapGestureRecognizer extends GestureRecognizer {
   ///  * [kPrimaryButton], the button this callback responds to.
   GestureDoubleTapCallback onDoubleTap;
 
+  /// Called when a pointer that will trigger a tap of a primary button has
+  /// stopped contacting the screen at a particular location, unless a previous
+  /// tap landed in proximity shortly before, in which case [onDoubleTap] and
+  /// [onDoubleTapUp] will be called.
+  ///
+  /// [onFirstTapUp] is called optimistically before winning the gesture arena,
+  /// the same pointer that triggered [onFirstTapUp] might be rejected afterwards.
+  /// Neither [onDoubleTap] or [onDoubleTapUp] is guaranteed to be called after
+  /// [onFirstTapUp].
+  GestureTapUpCallback onFirstTapUp;
+
+  /// Called when the user has tapped the screen with a primary button at the
+  /// same location twice in quick succession.
+  ///
+  /// This triggers when the pointer stops contacting the device after the
+  /// second tap, immediately after winning the gesture arenas of both taps.
```

Comment:
I'm guessing that this is needed for a "double tap and hold"?  So a quick tapdown, tapup, tapdown, then a long time and maybe some dragging, then this onDoubleTapUp fires when the user lifts their finger.  Does the regular onDoubleTap not fire in that situation?

Maybe explain this kind of stuff in the comments here or give an example, if you think it will be helpful.